### PR TITLE
Bug/ux reloading pages new

### PR DESCRIPTION
**Pull request by osala-eng**
## Description

Please include a summary of the change, e.g. what the new feature # is and/or what bug # it fixes. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes/Implements #(issue/feature)
Fixes page goes to login page on page reload even if you have already logged in.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [X] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] The commit message follows the convention of this project

### CODE REVIEW

**Comment by osala-eng** [comment]
> @osala-eng feature toggle change you mentioned on email is yet to be pushed right?

Yes

**Comment by janaka** [comment]
@cwang do you also want to review the feature toggle change? I've had a look and lgtm

**Comment by janaka** [comment]
@osala-eng is this rebased on latest main?

File: web/utils/handlers.py [hunk 1]
```diff
@@ -196,7 +216,7 @@ def _get_chat_spaces(feature: domain.FeatureKey) -> tuple[Optional[SpaceKey], Li
     elif feature.type_ == config.FeatureType.ASK_PUBLIC:
         return (
             None,
-            [domain.SpaceKey(config.SpaceType.PUBLIC, s_[0]) for s_ in list_public_space_group_members()],
+            [domain.SpaceKey(config.SpaceType.SHARED, s_[0]) for s_ in list_public_space_group_members()],
```
Comment (cwang) [inline web/utils/handlers.py:219]: Again wanted to double check if this is correct (from `PUBLIC` to `SHARED`)?

**Comment by janaka** [comment]
@osala-eng 
When I login and hit refresh I get logged out and I get the following error in the console. I delete `.persisted` then stop started the app.
>2023-09-12 14:59:22,782 ERROR Failed to check if auto login is enabled: argument of type 'NoneType' is not iterable

**Comment by osala-eng** [comment]
> @osala-eng When I login and hit refresh I get logged out and I get the following error in the console. I delete `.persisted` then stop started the app.
> 
> > 2023-09-12 14:59:22,782 ERROR Failed to check if auto login is enabled: argument of type 'NoneType' is not iterable

Let me take a look.

File: web/utils/sessions.py [hunk 2]
```diff
@@ -106,3 +106,8 @@ def set_public_session(val: dict = None, key: SessionKeyNameForPublic = None) ->
 def get_public_session(key: SessionKeyNameForPublic = None) -> int | None:
     """Get the public session value."""
     return _get_session_value(SessionKeySubName.PUBLIC, key.name if key is not None else None)
+
+
+def get_public_session_id() -> str | None:
+    """Get the public session id."""
+    return _get_session_value(SessionKeySubName.PUBLIC, SessionKeyNameForPublic.SESSION.name)
```
Comment (cwang) [inline web/utils/sessions.py:113]: Is the enum `SessionKeyNameForPublic` already in main?

File: web/utils/handlers.py (hunk 1 above)
Comment (osala-eng) [reply web/utils/handlers.py:219]: Yes, this points to the `Indexes` for public spaces which are stored under `SHARED`.

File: web/utils/sessions.py (hunk 2 above)
Comment (osala-eng) [reply web/utils/sessions.py:113]: Yes

File: source/docq/manage_settings.py [hunk 3]
```diff
@@ -40,7 +40,8 @@ def _get_sqlite_file(user_id: int = None) -> str:
 def _get_settings(user_id: int = None) -> dict:
     log.debug("Getting settings for user %s", str(user_id))
     with closing(
-        sqlite3.connect(_get_sqlite_file(user_id), detect_types=sqlite3.PARSE_DECLTYPES)
+        # sqlite3.connect(_get_sqlite_file(user_id), detect_types=sqlite3.PARSE_DECLTYPES)
+        sqlite3.connect(_get_sqlite_file(), detect_types=sqlite3.PARSE_DECLTYPES) # FIXME: fix sqlite3.OperationalError: no such table: settings if user_id is not None
```
[the remark this answers is not in the document]
Comment (janaka) [reply source/docq/manage_settings.py:44]: I gues this is a work around given the FIXME comment? but this change will break a bunch of stuff surely

File: source/docq/config.py [hunk 4]
```diff
@@ -22,6 +24,7 @@ class FeatureType(Enum):
     ASK_SHARED = "Ask Shared Documents"
     ASK_PUBLIC = "Ask Public Documents"
     CHAT_PRIVATE = "General Chat"
+    AUTO_LOGIN = "Auto Login"
```
[the remark this answers is not in the document]
Comment (osala-eng) [reply source/docq/config.py:27]: I added this as a `feature` to enable or disable re-authenticating on page reload. Not for the public sessions.

File: source/docq/manage_settings.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (janaka) [reply source/docq/manage_settings.py:44]: Looking at this closer, it seems like even the original logic will fail for system settings. When you login as a different user. Maybe best to fix that up on my branch given it's dealing with org scope settings.

[the remark this answers is not in the document]
Comment (osala-eng) [reply source/docq/manage_settings.py:44]: Sorry, I missed this when rebasing the fix was applied by #80

**Comment by janaka** [comment]
@osala-eng I presume these are all changes fixes you are slipping into this PR rather than being strictly related to the Auth feature?

**Comment by osala-eng** [comment]
> @osala-eng I presume these are all changes fixes you are slipping into this PR rather than being strictly related to the Auth feature?

I've refactored some code after merging #95  to avoid having duplicates

**Comment by osala-eng** [comment]
> The error has gone now. I still get logged out when I refresh.

This is disabled by default, You have to enable `Auto Login` in `Admin Settings`

**Comment by janaka** [comment]
> > The error has gone now. I still get logged out when I refresh.
> 
> This is disabled by default, You have to enable `Auto Login` in `Admin Settings`

seems to be enabled by default for me.

**Comment by osala-eng** [comment]
> > > The error has gone now. I still get logged out when I refresh.
> > 
> > 
> > This is disabled by default, You have to enable `Auto Login` in `Admin Settings`
> 
> seems to be enabled by default for me.

You have to click the save button on the `Admin Settings` page to make the `system settings` defined on a new instance of docq.

**Comment by osala-eng** [comment]
@janaka Should I make this enabled by default if system settings are not defined?

**Comment by janaka** [comment]
There's something weird doing on when booting up a new instance on the app. 

- setting page show Auto Login in the list, which I take as enabled. If it's disabled by default why is it appearing here? 

- Hitting save on the settings page is throwing an error in the console. I've tried removing Auto Login and re-adding. Still get the error.

If you are free let's jump on a call an hash out.

**Comment by janaka** [comment]
I created #99 to handle default setting issue. 

Note: once auto login has been enabled and settings saved, the user needs to logout and login for this particular feature to take effect. When it has, hitting refresh will not log the user out.

**Comment by janaka** [comment]
@osala-eng Sound like should be SameSite=strict. I don't think a auth token cookie should be sharable in the way lax does it I've understood it correctly.

**Review by janaka — APPROVED** [review]

File: web/utils/handlers.py (hunk 1 above)
Comment (cwang) [reply web/utils/handlers.py:219]: I don't understand this, why did it become incorrect after this change?

File: web/utils/sessions.py (hunk 2 above)
Comment (cwang) [reply web/utils/sessions.py:113]: I have some opinions about these names in general but it's okay for now.
Also I think this change should've belonged to another PR?

File: source/docq/config.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cwang) [reply source/docq/config.py:27]: This is really not ideal because it abuses the actual meaning of the definition here.

**Review by cwang — CHANGES_REQUESTED** [review]
Blocking temporarily so it's not merged for now before I have another look.

**Comment by cwang** [comment]
Is 32 really necessary, or just easy for us to make it so to satisfy the underlying crypto lib? 
Screenshot 2023-09-13 at 15 03 57

**Comment by janaka** [comment]
> Chat (api calls to openai) seems to be broken, probably not because of this PR.

note to help keep track: as suspected this was caused by a missing LlamaIndex dependency (NLTK). It's unclear how this got out of sync. Was fixed by bumping LlamaIndex to the latest stable version, removing the LlamaIndex, then adding again.

**Comment by osala-eng** [comment]
Reapplied this in #111

**Pull request closed without merging**